lattices/src/lib.rs

```diff
@@ -40,6 +42,61 @@ pub trait Lattice: Sized + Merge<Self> + LatticeOrd + NaiveLatticeOrd + IsBot +
 #[sealed]
 impl<T> Lattice for T where T: Sized + Merge<Self> + LatticeOrd + NaiveLatticeOrd + IsBot + IsTop {}
 
+/// Alias trait for groups.
+#[sealed]
+pub trait Group<T>: Addition<T> + Zero<T> + Inverse<T> {}
+
+/// Trait for Group Inverse.
+pub trait Inverse<T> {
+    /// Returns the inverse of the group element.
+    fn inverse(&self) -> Self;
+}
+
+/// Alias trait for semirings.
+#[sealed]
+pub trait Semiring<T>: Addition<T> + Multiplication<T> + Zero<T> + One<T> {}
```

Comment:
Add a
`impl<This, T> Semiring<T> for This where This: Addition<T> + Multiplication<T> + Zero<T> + One<T> {}`

Reply:
Fixed!